src/analysis/inner_product_space/basic.lean

```diff
@@ -1066,6 +1066,42 @@ begin
   simp only [sq, ← mul_div_right_comm, ← add_div]
 end
 
+section complex
+
+variables {V : Type*}
+[inner_product_space ℂ V]
+/--
+A complex polarization identity, with a linear map
+-/
+lemma inner_map_polarization (T : V →ₗ[ℂ] V) (x y : V):
+  ⟪ T y, x ⟫_ℂ = (⟪T (x + y) , x + y⟫_ℂ - ⟪T (x - y) , x - y⟫_ℂ +
+    complex.I * ⟪T (x + complex.I • y) , x + complex.I • y⟫_ℂ -
+    complex.I * ⟪T (x - complex.I • y), x - complex.I • y ⟫_ℂ) / 4 :=
+begin
+  simp only [map_add, map_sub, inner_add_left, inner_add_right, linear_map.map_smul,
+             inner_smul_left, inner_smul_right, complex.conj_I, ←pow_two, complex.I_sq,
+             inner_sub_left, inner_sub_right, mul_add, ←mul_assoc, mul_neg, neg_neg,
+             sub_neg_eq_add, one_mul, neg_one_mul, mul_sub, sub_sub],
+  ring,
+end
+
+/--
+If `⟪T x, x⟫_ℂ = 0` for all x, then T = 0.
+-/
+lemma inner_map_self_eq_zero (T : V →ₗ[ℂ] V) :
+  (∀ (x : V), ⟪T x, x⟫_ℂ = 0) ↔ T = 0 :=
+begin
+  split,
+  intro hT,
+  ext x,
+  simp only [linear_map.zero_apply, ← inner_self_eq_zero, inner_map_polarization, hT],
+  norm_num,
+  intros hT x,
+  simp only [eq_self_iff_true, inner_zero_left, linear_map.zero_apply, hT],
```

Comment:
```suggestion
  split,
  { intro hT,
    ext x,
    simp only [linear_map.zero_apply, ← inner_self_eq_zero, inner_map_polarization, hT],
    norm_num },
  { rintro rfl x,
    simp only [linear_map.zero_apply, inner_zero_left] }
```
When multiple goals are generated (such as here with the use of `split`), we normally break up the code into blocks delimited by braces like this. I also added an extra trick: with `rintro`, you can use the special token `rfl` to immediately substitute the hypothesis being introduced. So here we introduce `T = 0` and immediately substitute it everywhere we can.

Reply:
The braces for multiple goals sounds like good practice; I'll incorporate that going forward.

`rintro` is brand new to me, but I think I should have been using it in many places!